src/platform_impl/apple/appkit/window.rs

```diff
@@ -360,8 +360,34 @@ declare_class!(
     }
 );
 
-impl WinitWindow {
-    pub(super) fn id(&self) -> WindowId {
-        WindowId::from_raw(self as *const Self as usize)
+declare_class!(
+    #[derive(Debug)]
+    pub struct WinitPanel;
+
+    unsafe impl ClassType for WinitPanel {
+        #[inherits(NSWindow, NSResponder, NSObject)]
+        type Super = NSPanel;
+        type Mutability = mutability::MainThreadOnly;
+        const NAME: &'static str = "WinitPanel";
+    }
+
+    impl DeclaredClass for WinitPanel {}
+
+    unsafe impl WinitPanel {
+        #[method(canBecomeMainWindow)]
+        fn can_become_main_window(&self) -> bool {
+            trace_scope!("canBecomeMainWindow");
+            true
+        }
+
+        #[method(canBecomeKeyWindow)]
+        fn can_become_key_window(&self) -> bool {
+            trace_scope!("canBecomeKeyWindow");
+            true
+        }
```

Comment:
Nit: I think I'd prefer to avoid overriding these?

Reply:
Without `canBecomeKeyWindow` window doesn't receive keyboard inputs. Not sure if the `canBecomeMainWindow` one is needed

Reply:
But it says [here](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/WinPanel/Concepts/UsingPanels.html#//apple_ref/doc/uid/20000224) that panels can become the key window?

Reply:
Ok found it. `canBecomeKeyWindow` is needed if using `.with_decorations(false)`. this is what I need for my use case so I tested with it

Reply:
Specifically absence of `NSWindowStyleMask::Titled` makes it not accept keyboard events

Reply:
Cool, thanks for the investigation